src/field/field2d.cxx

```diff
@@ -535,26 +535,36 @@ Field2D pow(BoutReal lhs, const Field2D &rhs, REGION rgn) {
   return result;
 }
 
-#if CHECK > 0
-/// Check if the data is valid
-void checkData(const Field2D &f, REGION region) {
-  if(!f.isAllocated()) {
-    throw BoutException("Field2D: Operation on empty data\n");
-  }
-
+// Internal routine to avoid ugliness with interactions between CHECK
+// levels and UNUSED parameters
 #if CHECK > 2
+void checkData_is_finite_on_region(const Field2D &f, REGION region) {
```

Comment:
I'm not sure about this; possibly replaces one kind of ugliness with another. Pedantic point: probably should be `checkDataIsFiniteOnRegion`.

Reply:
Good catch on the name, thanks!

Yes, I'm not very happy with it. I think the other alternatives were:

a)
```cpp
#if CHECK > 2
void checkData(const Field2d &f, REGION region)
#elif CHECK > 0
void checkData(const Field2d &f, REGION UNUSED(region))
#endif
{
```
b)
```cpp
void checkData(const Field2d &f, 
#if CHECK > 2
               REGION region)
#elif CHECK > 0
               REGION UNUSED(region))
#endif
{
```
Or of course, just don't warn on unused parameters (by default). It's not really helpful for end-users, but developers should really have it on.

Reply:
Maybe only not warn about unused parameters when compiled with checks?